### PR TITLE
chore(deps): updated frontend's dependencies

### LINKED ISSUE #75: Migrate to Parcel v2
https://parceljs.org/getting-started/migration/

### PR DESCRIPTION
**Pull request by DuCanhGH**
**Brief Description of the Changes**
This PR updates multiple dependencies to latest, including Parcel, and adjusts other code accordingly.

Closes #75 
Closes #76.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -2,41 +2,41 @@
...
+        "@fullhuman/postcss-purgecss": "5.0.0",
+        "@types/humanize-duration": "3.27.1",
+        "@types/node": "20.1.3",
+        "@types/react": "18.2.6",
+        "autoprefixer": "10.4.14",
+        "parcel": "2.8.3",
+        "postcss": "8.4.23",
+        "prettier": "2.8.8",
+        "tailwindcss": "3.3.2",
+        "typescript": "5.0.4"
```
Comment (natsukagami) [inline frontend/package.json:19]: Why the fixed versions rather than `^` semver-compatible versions? This breaks `yarn upgrade`/`yarn upgradeInteractive`

**Comment by natsukagami** [comment]
You would probably need something like https://github.com/natsukagami/kjudge/pull/76/files#diff-173ca6f64f98bd376d0291a11f5abefbfacb245a3764ccb9743b50e565a7d52e to avoid parcel/htmlnano from doing something nasty to the Go template syntax

File: frontend/package.json (hunk 1 above)
Comment (DuCanhGH) [reply frontend/package.json]: @natsukagami fixed in [22f52b2](https://github.com/natsukagami/kjudge/pull/80/commits/22f52b20f24805104ee1006b52d521e74a8484cb). I just think that pinning dependencies might be better (having another `node-ipc` won't be fun...).

File: cmd/kjudge/main.go [hunk 2]
```diff
@@ -18,7 +18,7 @@ import (
 
 var (
 	dbfile      = flag.String("file", "kjudge.db", "Path to the database file.")
-	sandboxImpl = flag.String("sandbox", "isolate", "The sandbox implementation to be used (isolate, raw). If anything other than 'raw' is given, isolate is used.")
+	sandboxImpl = flag.String("sandbox", "raw", "The sandbox implementation to be used (isolate, raw). If anything other than 'raw' is given, isolate is used.")
```
Comment (natsukagami) [inline cmd/kjudge/main.go]: Please revert, `raw` is an unsafe default option

File: frontend/ts/scoreboard/index.tsx [hunk 3]
```diff
@@ -1,7 +1,10 @@
+import "regenerator-runtime/runtime";
+
 import { render } from "preact";
 import { useState, useEffect } from "preact/hooks";
-import FlipMove from "react-flip-move";
-import "regenerator-runtime/runtime";
+import FlipMoveModule from "react-flip-move";
+
+const FlipMove = FlipMoveModule as any;
```
Comment (natsukagami) [inline frontend/ts/scoreboard/index.tsx:7]: Why this change?

File: cmd/kjudge/main.go (hunk 2 above)
Comment (DuCanhGH) [reply cmd/kjudge/main.go]: @natsukagami that was for debugging because I don't have `isolate` installed 😅

File: frontend/html/root.html [hunk 4]
```diff
@@ -1,6 +1,8 @@
 <!DOCTYPE html>
 
 <head>
+    <meta charset="utf-8">
+    <link rel="icon" href="../img/confused_kanna.png" type="image/x-icon" sizes="any">
```
Comment (natsukagami) [inline frontend/html/root.html:5]: Wtf

**Comment by natsukagami** [comment]
OK we'll merge and fix bugs we'll run into later :)
